File: cs12b/pa3/DictionaryTest.java
public class DictionaryTest{

   public static void main(String[] args){
      
      String v;
      Dictionary A = new Dictionary();
      A.insert("a","apple");
      A.insert("b","bat");
      A.insert("c","cat");
      A.insert("d","dog");
      A.insert("e","elephant");
      A.insert("f","frog");
      A.insert("g","goat"); 
      System.out.println(A);


      v = A.lookup("a"); 
      System.out.println("key=a "+(v==null?"not found":("value="+v)));
      v = A.lookup("b");
      System.out.println("key=b "+(v==null?"not found":("value="+v)));
      v = A.lookup("g");
      System.out.println("key=g "+(v==null?"not found":("value="+v)));
      v = A.lookup("z"); 
      System.out.println("key=z "+(v==null?"not found":("value="+v)));
      System.out.println();
   
      // A.insert("a","ape");  // causes KeyCollisionException
       
       System.out.println();

       A.delete("e");
       A.delete("g");
       A.delete("b");
       System.out.println(A);

      // A.delete("h");  // causes KeyNotFoundException

       System.out.println(A.isEmpty());
       System.out.println(A.size());
       A.makeEmpty();
       System.out.println(A.isEmpty());
       System.out.println(A);

   }
}

File: cs12m/lab5/DictionaryTest.c
#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h"

#define MAX_LEN 180

int main(int argc, char* argv[]){

      char* v;
      Dictionary A = newDictionary();
      insert(A,"a","apple");
      insert(A,"b","bat");
      insert(A,"c","cat");
      insert(A,"d","dog");
      insert(A,"e","elephant");
      insert(A,"f","frog");
      insert(A,"g","goat");
      printDictionary(stdout, A);


      v=lookup(A,"a");
      printf("key=\"%s\" %s\"%s\"\n", "a", (v==NULL?"not found ":"value="), "apple");
      v=lookup(A,"g");
      printf("key=\"%s\" %s\"%s\"\n", "g", (v==NULL?"not found ":"value="), "goat");

      // A.insert(A,"a","ape");  // causes KeyCollisionException

      // A.delete(A,"h");  // causes KeyNotFoundException

      delete(A, "a");
      delete(A, "c");
      delete(A, "g");

      printDictionary(stdout, A);

      v=lookup(A,"a");
      printf("key=\"%s\" %s\"%s\"\n", "a", (v==NULL?"not found ":"value="), "apple");
      v=lookup(A,"z");
      printf("key=\"%s\" %s\"%s\"\n", "g", (v==NULL?"not found ":"value="), "goat");


   printf("%s\n", (isEmpty(A)?"true":"false"));
   printf("%d\n", size(A));
   makeEmpty(A);
   printf("%s\n", (isEmpty(A)?"true":"false"));
   freeDictionary(&A);

   return(EXIT_SUCCESS);
}
